analysis/staticAnalysisReport.Rmd

```diff
@@ -0,0 +1,171 @@
+---
+title: "Eval Static Analysis"
+author: "Chakshu Goyal"
+date: "07/06/2018"
+output:
+  html_document: default
+  pdf_document: default
+---
+
+## 
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+library(ggplot2)
+library(tidyverse)
+library(tools)
+#knitr::kable(head(arrange(evalCntPerPkg, desc(freq)) , 20) , col.names = c("Package" , "Number of evals"))
+#knitr::kable(head(arrange(outlyingRows, desc(freq)) , 5) , col.names = c("Package" , "Number of evals"))
+#lineCountPerPkg <- 
+#  lineCountPerFile %>%
+#    group_by(pkg) %>%
+#      summarise(lineCount = sum(lineCount), charCount = sum(charCount))
+#lineNdEvalCountPerPkg <- merge(lineCountPerPkg, evalCntPerPkg)
+#evalCntPerPkgLeft <- evalCntPerPkg[ !(evalCntPerPkg$pkg %in% lineCountPerPkg$pkg) , ]
+#evalCntPerPkgLeft
+```
+
+## Load the data
+Usage of the function eval in R-packages.
+To get evals.csv, run - Rscript evalCrawler.R /mnt/nvme0/R/CRAN-extracted
+```{r}
+evaldf <- read_csv("evals.csv")
+```
+
+## Plot occurences of evals in packages
+```{r}
+ggplot(evaldf , aes(pkg)) + 
+  geom_bar() + 
+  theme(axis.text.x = element_blank()) + 
+  labs(y="Number of evals", x="packages", title="Figurative distribution of the usage of eval in various packages")
+```
+
+Top 20 packages using eval
+```{r}
+evalCntPerPkg <- dplyr::count(evaldf, pkg)
+knitr::kable(
+  evalCntPerPkg %>%
+  top_n(20, wt=n) %>%
+  arrange(desc(n)))
+```
+
+Eval distribution in packages - density and histogram plots
+```{r}
+ggplot(evalCntPerPkg, aes(n)) + 
```

Comment:
If you find the density important, merge it to histogram.
Check the formatting.

Reply:
Acknowledged!